### DISCUSSION TITLE
FloatingNodeRuleViolation cause

### DISCUSSION THREAD

**Discussion by Abbas111977** [SpiceSharp/SpiceSharp#204] [Q&A] [answered]
Dears, Iam getting the rules violation exception while doing OP simulation for resistors and diodes, what is the meaning of FloatingNodeRuleViolation  and also VariablePresenceRule rules violation? is there any documentation that explain more about the validation rules used in SpiceSharp ? 
thanks in advance

**Comment by svenboulanger** [comment, marked as answer]
Hi Abbas,

The floating node rule violation means that SpiceSharp detected that a node does not have a resistive path to ground, which will usually result in a singular matrix.

The variable presence rule means that there should be at least one occurence of a certain variable, of which there is only one such example and that is the ground node. Please make sure at least one node of your circuit is ground ("0" or "gnd").

I think both may be linked actually. If you don't have a ground node anywhere in your circuit, then SpiceSharp will detect that the whole circuit is floating compared to ground, and may also give a floating node rule violation.

Yours sincerely,
Sven

**Reply by Abbas111977** [reply]
Many thanks for your kind, fast response 
Best Regards
